example/quicklook.cpp

```diff
@@ -0,0 +1,207 @@
+//
+// Copyright (c) 2022 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CppAlliance/url
+//
+
+//[snippet_headers_1
+#include <boost/url.hpp>
+//]
+
+#if 0
+//[snippet_headers_2
+#include <boost/url/src.hpp>
+//]
+#endif
+
+#include <iostream>
+
+//[snippet_headers_3
+#include <boost/url.hpp>
+using namespace boost::urls;
+//]
+
+void
+using_url_views() {
+    //[snippet_parsing_1
+    string_view url_str = "https://user:pass@www.example.com:443/path/to/my%2dfile.txt?id=42&name=John%20Doe#page%20anchor";
+    //]
+
+    {
+        //[snippet_parsing_2
+        result<url_view> r = parse_uri( url_str );
+        //]
+    }
+
+    {
+        //[snippet_parsing_3
+        url_view u = parse_uri( url_str ).value();
+        //]
+    }
+
+    {
+        //[snippet_parsing_4
+        try
+        {
+            url_view u = parse_uri(url_str).value();
+        }
+        catch (std::invalid_argument const&)
+        {
+            // this happens
+        }
+        //]
+    }
+
+    {
+        //[snippet_parsing_5
+        result<url_view> r = parse_uri( url_str );
+        if (r.has_value())
+        {
+            url_view u = r.value();
+        }
+        else
+        {
+            // this happens
+        }
+        //]
+    }
+
+    url_view u = parse_uri( url_str ).value();
+
+    //[snippet_accessing_1
+    std::cout <<
+        "scheme    : " << u.scheme()            << '\n' <<
+        "authority : " << u.encoded_authority() << '\n' <<
+        "path      : " << u.encoded_path()      << '\n' <<
+        "query     : " << u.encoded_query()     << '\n' <<
+        "fragment  : " << u.encoded_fragment()  << '\n';
+    //]
+
+    //[snippet_decoding_1
+    std::cout <<
+        "query    : " << u.query()            << '\n' <<
```

Comment:
All these functions using const_string give me two warnings from static analysis. 

```
Slicing object from type 'const_string' to 'basic_string_view<char>' discards 32 bytes of state
```

and 

```
Use pointer or reference to avoid slicing from "const_string" to "basic_string_view<char>".
```